enterprise_subsidy/apps/api/v1/tests/test_views.py

```diff
@@ -687,15 +688,15 @@ def test_retrieve_invalid_uuid(self):
     # Uncomment this later once we have segment events firing.
     # @mock.patch('enterprise_subsidy.apps.api.v1.event_utils.track_event')
     @mock.patch("enterprise_subsidy.apps.subsidy.models.Subsidy.enterprise_client")
-    @mock.patch("enterprise_subsidy.apps.api_client.enterprise_catalog.EnterpriseCatalogApiClient.get_course_price")
-    def test_create(self, mock_get_course_price, mock_enterprise_client):
+    @mock.patch("enterprise_subsidy.apps.subsidy.models.Subsidy.price_for_content")
+    def test_create(self, mock_price_for_content, mock_enterprise_client):
         """
         Test create Transaction, happy case.
         """
         url = reverse("api:v1:transaction-list")
         test_enroll_enterprise_fulfillment_uuid = "test-enroll-reference-id"
         mock_enterprise_client.enroll.return_value = test_enroll_enterprise_fulfillment_uuid
-        mock_get_course_price.return_value = "100.00"
+        mock_price_for_content.return_value = 10000
```

Comment:
This was a pretty sneaky bug that caused tests to be non-deterministic.  Basically, if a function is cached (e.g. via the `@lru_cache()` decorator), it will cache between unit tests unless we mock the function itself which overrides the cache.

Reply:
👍 good catch.